dftimewolf/lib/exporters/s3_to_gcs.py

```diff
@@ -0,0 +1,155 @@
+from dftimewolf.lib.state import DFTimewolfState
+
+
+class S3ToGCSCopy(module.ThreadAwareModule):
+  """AWS S3 objects to GCP GCS.
+
+  Attributes:
+    aws_region (srt): AWS region (for account.AWSAccount creation).
+    dest_project (gcp_project.GoogleCloudProject): Destination project with the
+      destination GCS bucket.
+    dest_project_name (str): Name of the destination project. used to create
+      the dest_project.
+    dest_bucket (str): Destination bucket in the GCP project.
+    s3_objects (List[str]): Objects to be copied.
+    filter (str): regex filter for objects to copy - Useful when the files are
+      from a previous module.
+  """
+
+  def __init__(self,
+               state: DFTimewolfState,
+               name: Optional[str]=None,
+               critical: bool=False) -> None:
+    """Initializes a copy of objects from AWS S3 to a GCS bucket.
+
+    Args:
+      state (DFTimewolfState): recipe state.
+      name (Optional[str]): The module's runtime name.
+      critical (Optional[bool]): True if the module is critical, which causes
+          the entire recipe to fail if the module encounters an error.
+    """
+    super(S3ToGCSCopy, self).__init__(
+        state, name=name, critical=critical)
+    self.aws_region: str = ''
+    self.dest_project_name: str = ''
+    self.dest_project: gcp_project.GoogleCloudProject = ''
+    self.dest_bucket: str = ''
+    self.filter: Any = None
+
+  # pylint: disable=arguments-differ
+  def SetUp(self,
+            aws_region: str,
+            dest_project: str,
+            dest_bucket: str,
+            s3_objects: str='',
+            object_filter: str='') -> None:
+    """Sets up a copy operation from AWS S3 to GCP GCS.
+
+    AWS objects to copy are sourced from either the state, or passed in here.
+    Args:
+      aws_region (str): The AWS region (for account.AWSAccount creation).
+      dest_project (str): The destination GCP project.
+      dest_bucket (str): The destination GCP bucket.
+      s3_objects (str): Comma separated list of objects to copy from S3. Each
+        should be of the form 's3://bucket-name/path/to/object'
+      object_filter (str): regex filter for objects to copy - Useful when the
+        files are from a previous module but not all should be transferred.
+    """
+    self.aws_region = aws_region
+    self.dest_project_name = dest_project
+    self.dest_bucket = dest_bucket
+    self.dest_project = gcp_project.GoogleCloudProject(self.dest_project_name)
+    self.filter = object_filter
+
+    if s3_objects:
+      for obj in s3_objects.split(','):
+        self.state.StoreContainer(containers.AWSS3Object(obj))
+
+  def PreProcess(self) -> None:
+    """Prep work for copying objects from S3 to GCS - Bucket creation."""
+    # Check if the destination bucket exists. If not, create it.
+    if not self.dest_bucket:
+      self.ModuleError('No destination GCP bucket specified', critical=True)
+    if self.dest_bucket not in \
```

Comment:
Style nit:  This is another case I would probably put on two lines for readability